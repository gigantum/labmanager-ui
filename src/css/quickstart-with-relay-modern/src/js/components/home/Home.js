import React, { Component } from 'react'
import {
  QueryRenderer,
  graphql
} from 'react-relay'
import environment from '../../createRelayEnvironment'

import DatasetsLabbooksContainer from '../datasetsLabbooks/DatasetsLabbooksContainer';
//import ListPage from './ListPage'

const LabbookQuery = graphql`query HomeQuery($first: Int!){localLabbooks(first:$first){edges{node{name}}}}`


export default class Home extends Component {

  login() {
    this.props.auth.login();
  }
  render() {
    console.log(this)
    const { isAuthenticated } = this.props.auth;
    console.log(isAuthenticated())
    return (
      <div>
      <div>
        {
        isAuthenticated() && (
        <QueryRenderer
          environment={environment}
          query={LabbookQuery}
          variables={{
            first: 4
          }}
          render={({error, props}) => {
            console.log(props)
            if (error) {
              return <div>{error.message}</div>
            } else if (props) {
              return <div></div>
            }
            return <div>Loading</div>
          }}
        />)
      }
      {
        !isAuthenticated() && (
            <h4>
              You are not logged in! Please{' '}
              <a
                style={{ cursor: 'pointer' }}
                onClick={this.login.bind(this)}
              >
                Log In
              </a>
              {' '}to continue.
            </h4>
          )
      }
      </div>
      <DatasetsLabbooksContainer />
    </div>
    )
  }
}
