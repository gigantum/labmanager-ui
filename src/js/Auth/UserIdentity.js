//vendor
import {
  graphql,
} from 'react-relay'
//mutations
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//environment
import {fetchQuery} from 'JS/createRelayEnvironment';

const userIdentityQuery = graphql`
  query UserIdentityQuery{
    userIdentity{
      id
      username
      email
      givenName
      familyName
      isSessionValid
    }
  }
`;

const UserIdentity = {
  getUserIdentity: () => {

    return new Promise((resolve, reject) =>{

      let fetchData = function(){

        fetchQuery(userIdentityQuery(), {}).then((response, error) => {

          if(response){
            resolve(response)
          }else{

            reject(response)
          }
        }).catch((error) =>{

          reject(error)
        })
      }

      fetchData()
    })
  },
  removeUserIdentity: () => {
    RemoveUserIdentityMutation(()=>{
      console.log('removed user')
    })
  }

}

export default UserIdentity
