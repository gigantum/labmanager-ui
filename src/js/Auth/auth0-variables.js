export const AUTH_CONFIG = {
  domain: 'calum.auth0.com',
  clientId: '9wFN5CZA4Z9xa1r8KZIZbevpjHG7bAab',
  callbackUrl: window.location.origin + '/callback'
}
