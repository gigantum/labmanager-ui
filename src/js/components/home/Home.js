import React, { Component } from 'react'
import Dashboard from 'Components/dashboard/Dashboard';
import Login from 'Components/login/Login';

export default class Home extends Component {
  //login for Auth0 function
  constructor(props){
    super(props)

    this.footerWorkerCallback = this.footerWorkerCallback.bind(this)
  }
  login() {
    this.props.auth.login();
  }

  footerWorkerCallback(worker, filepath){
    this.props.footerWorkerCallback(worker, filepath);
  }
  render() {
    const { isAuthenticated } = this.props.auth;
    const { forceLoginScreen, loadingRenew } = this.props;

    return (
      <div className="Home">
        {
          isAuthenticated() && !forceLoginScreen && (
            <Dashboard
              auth={this.props.auth}
              footerWorkerCallback={this.footerWorkerCallback}
              match={this.props.match}
              history={this.props.history}
            />
          )
        }

        {
          (!isAuthenticated() || forceLoginScreen) && (
              <Login
                auth={this.props.auth}
                forceLoginScreen={forceLoginScreen}
                loadingRenew={loadingRenew}
              />
            )
        }

    </div>
    )
  }
}
