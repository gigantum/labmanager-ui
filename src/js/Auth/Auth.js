import history from 'JS/history';
import auth0 from 'auth0-js';
import { AUTH_CONFIG } from './auth0-variables';
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//store
import store from 'JS/redux/store'


export default class Auth {
  auth0 = new auth0.WebAuth({
    domain: AUTH_CONFIG.domain,
    clientID: AUTH_CONFIG.clientId,
    redirectUri: AUTH_CONFIG.callbackUrl,
    audience:  AUTH_CONFIG.audience,
    responseType: 'token id_token',
    scope: 'openid profile email user_metadata'
  });

  constructor() {
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.handleAuthentication = this.handleAuthentication.bind(this);
    this.isAuthenticated = this.isAuthenticated.bind(this);

  }

  login() {
    this.auth0.authorize();
  }

  handleAuthentication() {

    this.auth0.parseHash((err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {

        this.setSession(authResult);

      } else if (err) {
        console.error(err);
      //  alert(`Error: ${err.error}. Check the console for further details.`);
      }
    });
  }

  setSession(authResult) {

    // Set the time that the access token will expire at
    let expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
    localStorage.setItem('access_token', authResult.accessToken);
    localStorage.setItem('id_token', authResult.idToken);
    localStorage.setItem('expires_at', expiresAt);
    localStorage.setItem('family_name', authResult.idTokenPayload.family_name);
    localStorage.setItem('given_name', authResult.idTokenPayload.given_name);
    localStorage.setItem('email', authResult.idTokenPayload.email);
    localStorage.setItem('username', authResult.idTokenPayload.nickname);
    //redirect to labbooks when user logs in

    let storrageRoute = sessionStorage.getItem('CALLBACK_ROUTE')
    let route = storrageRoute !== '' ? storrageRoute : `/labbooks`

    history.replace(route)
  }

  logout() {

    localStorage.removeItem('access_token');
    localStorage.removeItem('id_token');
    localStorage.removeItem('expires_at');
    localStorage.removeItem('family_name')
    localStorage.removeItem('given_name');
    localStorage.removeItem('email');
    localStorage.removeItem('username');
    RemoveUserIdentityMutation(()=>{
      //redirect to root when user logs out
      history.replace('/');
    })
  }

  isAuthenticated() {
    // Check whether the current time is past the
    // access token's expiry time
    let expiresAt = JSON.parse(localStorage.getItem('expires_at'));

    return new Date().getTime() < expiresAt;
  }
}
