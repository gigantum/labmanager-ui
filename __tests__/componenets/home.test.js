
import Home from 'Components/home/Home';
import React from 'react';
import renderer from 'react-test-renderer';
import Auth from 'JS/Auth/Auth';
import { Router } from "react-router";
const auth = new Auth();
auth.isAuthenticated = function(){return false};


test('Test Home Rendering', () => {
      const component = renderer.create(

          <Home auth={auth}/>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})
