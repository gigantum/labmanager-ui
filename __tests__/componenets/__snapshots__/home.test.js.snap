// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Rendering 1`] = `
<div
  className="Home"
>
  <div
    className="Login"
  >
    <div
      className="Login__panel flex flex--column justify--space-around"
    >
      <button
        className="Login__button"
        onClick={[Function]}
      >
        Log In
      </button>
    </div>
  </div>
</div>
`;
